contracts/FlowToken.cdc

```diff
@@ -83,6 +84,64 @@ pub contract FlowToken: FungibleToken {
                 FlowToken.totalSupply = FlowToken.totalSupply - self.balance
             }
         }
+
+        /// Get all the Metadata Views implemented by FlowToken
+        ///
+        /// @return An array of Types defining the implemented views. This value will be used by
+        ///         developers to know which parameter to pass to the resolveView() method.
+        ///
+        pub fun getViews(): [Type]{
+            return [Type<FungibleTokenMetadataViews.FTView>(),
+                    Type<FungibleTokenMetadataViews.FTDisplay>(),
+                    Type<FungibleTokenMetadataViews.FTVaultData>()]
+        }
+
+        /// Get a Metadata View from FlowToken
+        ///
+        /// @param view: The Type of the desired view.
+        /// @return A structure representing the requested view.
+        ///
+        pub fun resolveView(_ view: Type): AnyStruct? {
+            switch view {
+                case Type<FungibleTokenMetadataViews.FTView>():
+                    return FungibleTokenMetadataViews.FTView(
+                        ftDisplay: self.resolveView(Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
+                        ftVaultData: self.resolveView(Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+                    )
+                case Type<FungibleTokenMetadataViews.FTDisplay>():
+                    let media = MetadataViews.Media(
+                            file: MetadataViews.HTTPFile(
+                            url: "https://assets.website-files.com/5f6294c0c7a8cdd643b1c820/5f6294c0c7a8cda55cb1c936_Flow_Wordmark.svg"
```

Comment:
Do we know where this is hosted? It'd be great to get this pinned in IPFS or some other Web3 native solution.

Reply:
We might just store it on chain in xml format since it is small

Reply:
Anybody have any ideas about which metadata view would be the best to use for storing an entire XML file? This one would only be about 2KB so we can probably just store it all on chain, but I don't know which one to use or if we have to create a new view for it

Reply:
@joshuahannan `HTTPFile` with `data` URI ?

Reply:
So I would just save the entire XML file as a `String` in the URI field?

Reply:
So long as you encode the start with the right encoding that should work, I think. Only returning the string won't tell platforms how to interpret the data being returned

Reply:
it will be something like: 

```data:text/xml;base64,iVBORw0KGgoAAA
ANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4
//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU
5ErkJggg==```